backend/src/v1/data/repositories/wordRepository.ts

```diff
@@ -1,80 +1,88 @@
-import { Word } from "../models/models";
-import { db } from "../../../db/database";
-import { DeleteResult, InsertResult } from "kysely";
-export interface WordRepository {
-    getWords(languageCode?: string): Promise<Word[]>;
-    getWord(id: number): Promise<Word>;
-    insertWord(data: WordInsert): Promise<{ insertedRows: string; }>;
-    updateWord(data: WordUpdate): Promise<{ updatedRows: string; }>;
-    deleteWord(id: number): Promise<{ deletedRowsCount: string; }>;
+import { db } from '../../../db/database.js';
+
+type Word = {
+	language_code: string;
+	word_id: number;
+	word_text: string;
+};
+
+type InsertData = {
+	text: string;
+	languageCode: string;
+};
+
+type UpdateData = {
+	id: number;
+	text: string;
+	languageCode: string;
 };
 
-const WordRepositoryImpl: WordRepository = {
-    getWords: async (languageCode: string): Promise<Word[]> => {
-        let query = db.selectFrom("word").selectAll();
-        if (languageCode) {
-            query.where("language_code", "=", languageCode);
-        }
-        let result = await query.execute();
-        return result.map((entity) => {
-            return {
-                id: entity.word_id,
-                text: entity.word_text,
-                languageCode: entity.language_code
-            };
-        });
-    },
-    getWord: async (id: number): Promise<Word> => {
-        let data = await db.selectFrom("word")
-            .selectAll()
-            .where("word.word_id", "=", id)
-            .executeTakeFirstOrThrow();
+type GetWords = (languageCode?: string) => Promise<Word[]>;
+type GetWord = (id: number) => Promise<Word>;
+type InsertWord = (data: InsertData) => Promise<{ insertedRows: string }>;
+type UpdateWord = (data: UpdateData) => Promise<{ updatedRows: string }>;
+type DeleteWord = (id: number) => Promise<{ deletedRowsCount: string }>;
+
+export const getWords: GetWords = async (languageCode) => {
+	const query = db
+		.selectFrom('word')
+		.selectAll()
+		.$if(languageCode !== undefined, (builder) =>
+			builder.where('word.language_code', '=', languageCode!),
+		);
+
+	try {
+		const result: Word[] = await query.execute();
+		return result;
+	} catch (error) {
+		if (typeof error === 'string') {
```

Comment:
Does Kysely throw strings? What's going on here?

Reply:
oh woops. this was leftover as i was experimenting with error handling. I actually meant to add some utils to help with that.